---
title: "CAF subtype analysis"
author: "Kevin Ryan"
date: "`r Sys.time()`"
bibliography: citations.bib
output: 
  github_document:
     toc: true
     toc_depth: 3 
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Introduction

Cancer-associated fibroblasts (CAFs) are a heterogeneous cell type found in the tumour microenvironment. They have a wide array of functions, and tend to be immunosuppressive and cancer-promoting. There have been many attempts to characterise subtypes of CAFs, with much transcriptomic analysis being carried out in the Mechta-Grigoriou lab in Institut Curie. They have identified 4 'subtypes' which can be separated based on the expression of different markers:

 -  S1: FAP^High^, CD29^Med-High^, α^SMAHigh^, PDPN^High^, PDGFRβ^High^
 -  S2: FAP^Neg^, CD29^Low^, αSMANeg-^Low^, PDPN^Low^, PDGFRβ^Low^
 -  S3: FAP^Neg-Low^, CD29^Med^, αSMA^Neg-Low^, PDPN^Low^, PDGFRβ^Low-Med^
 -  S4: FAP^Low-Med^, CD29^High^, αSMA^High^, PDPN^Low^, PDGFRβ^Med^

[@Pelon2020]

FACS gating strategies can be used to isolate these various subtypes. The Mechta-Grigoriou group have done this and have generated bulk RNA-sequencing data for the S1, S3 and S4 subtypes. They generated scRNA-sequencing data for the S1 subtype. This data was deposited on the European Genome Phenome Archive, and was accessed via a Data Transfer Agreement.

The following summarises the data obtained:

+---------------+---------------+-----------------------+-------------------------------+
| Subtype       | Total samples | Studies (Samples)     |  Notes                        |
+===============+===============+=======================+===============================+
| S1            |  28           | - EGAD00001003808 (16)| -  3808 has 12xJuxta-tumor    |
|               |               | - EGAD00001005744 (5) | -  5744 5 samples from LN     |
|               |               | - EGAD00001006144 (7) | -  Sorting vs spreading       |
+---------------+---------------+-----------------------+-------------------------------+
| S2            | 0             | N/A                   | N/A                           |
|               |               |                       |                               |
+---------------+---------------+-----------------------+-------------------------------+
| S3            | 14            | - EGAD00001004810 (14)| -  4810 has 11xJuxta-tumor    |
|               |               |                       | -  Ovarian                    |
+---------------+---------------+-----------------------+-------------------------------+
| S4            | 15            | - EGAD00001003808 (10)| -  3808 has 9xJuxta-tumor     |
|               |               | - EGAD00001005744 (5) | -  5744 5 samples from LN     |
+---------------+---------------+-----------------------+-----------------------+

With the juxta-tumour data, they got tumour and juxta-tumour data from the same patient. However, I have not been able to figure out whether they came from the same patient. Could probably use Optitype to determine HLA allele - match tumour and juxta tumour.

We also have scRNA-seq data for S1.

The data was processed using nf-core/rnaseq version `3.8.1` using the default parameters. STAR/Salmon were used for alignment/quantification. In-house CAF and tumour-associated normal (TAN) RNA-sequencing data were processed using nf-core/rnaseq version `3.1`, and STAR/Salmon were also used for alignment/quantification. 

We would expect our tumour-associated normal to be most like the S3 subtype (usually accumulate in juxta-tumours).

Combining RNA-sequencing datasets from different studies can be very challenging. We can expect batch effects to be present, so it might be possible to determine whether differences be observe are due to actual biological effects or technical artifacts. In addition, a recent study suggests that DESeq2 and edgeR (the most popular differential expression tools) experience large rates of false positives when used with large sample sizes [@Li2022]. One of the datasets (`EGAD00001006144`) was produced using stranded RNA-seq, whereas the other datasets were unstranded. This can lead to a lack of comparability of the datasets [@Zhao2020]. It may be necessary to drop this dataset from the analysis. All samples were prepared by poly(A) selection (use of oligo-dT).

## Preparation
### Create Sample File

Columns will be: Sample, Study, CAF_subtype, Tumor_Juxtatumor

*Here we will be combining data from 5 studies. To begin with, we will only include the metadata available for all studies (except for our unknown CAF subtype label). Breast cancer subtype is only available for certain studies and so is not included at this stage.*

There are also: ovarian cancer samples, EPCAM+ cells, samples prepared by spreading or spreading and samples from lymph nodes. For the time being, I will not consider them.

```{r message=FALSE}
suppressPackageStartupMessages(library(dplyr))
suppressPackageStartupMessages(library(stringr))
library(biomaRt)
suppressPackageStartupMessages(library(tximport))
library(DT)
suppressPackageStartupMessages(library(tidyverse))
suppressPackageStartupMessages(library(ggplot2))
library(cowplot)
suppressPackageStartupMessages(library(PCAtools))
library(dplyr)
suppressPackageStartupMessages(library(SummarizedExperiment))
library(DESeq2)
```

```{r read in metadata}
EGAD_4810 <- read.table("/home/kevin/Documents/PhD/CAF_data/EGAD00001004810/delimited_maps/Run_Sample_meta_info.map", 
                        sep = ";")
EGAD_4810_cancers <- str_split_fixed(EGAD_4810$V1, pattern = "=", n = 2)[,2]
EGAD_4810_keep <- which(EGAD_4810_cancers == "BC")
EGAD_4810_filtered <- EGAD_4810[EGAD_4810_keep,]
EGAD_4810_meta <- data.frame(
  Sample = str_split_fixed(EGAD_4810_filtered$V4, pattern = "=", n = 2)[,2],
  Study = "EGAD00001004810",
  Subtype = "S3",
  Tumor_JuxtaTumor = tolower(str_split_fixed(EGAD_4810_filtered$V3, pattern = " ", n = 2)[,2]),
  directory = "/home/kevin/Documents/PhD/CAF_data/nfcore_results/EGAD00001004810_nfcore_results/star_salmon",
  row.names = 1
)
EGAD_3808 <- read.table("/home/kevin/Documents/PhD/CAF_data/EGAD00001003808/meta_CAF-S1_S4_BC_47samples.txt", 
                        header = T, sep = "\t")
EGAD_3808_meta <- data.frame(
  Sample = EGAD_3808$Sample.Name,
  Study = "EGAD00001003808",
  Subtype = EGAD_3808$subset,
  Tumor_JuxtaTumor = EGAD_3808$Type, 
    directory = "/home/kevin/Documents/PhD/CAF_data/nfcore_results/EGAD00001003808_nfcore_results/star_salmon",
  row.names = 1
)
EGAD_6144 <- read.table("/home/kevin/Documents/PhD/CAF_data/EGAD00001006144/meta_7samples.txt", 
                        header = T,sep = "\t")
EGAD_6144_meta <- data.frame(
  Sample = paste("CAF_Culture_", EGAD_6144$Sample.Name, sep = ""),
  Study = "EGAD00001006144",
  Subtype = "S1",
  Tumor_JuxtaTumor = "tumor",
    directory = "/home/kevin/Documents/PhD/CAF_data/nfcore_results/EGAD00001006144_nfcore_results/star_salmon",
  row.names = 1
)
EGAD_5744 <- read.table("/home/kevin/Documents/PhD/CAF_data/EGAD00001005744/metaData_Pelon_et_al.txt", 
                        header =T, check.names = F)
EGAD_5744$Sample.Name <- gsub(pattern = "\\.", replacement = "-", x = EGAD_5744$Sample.Name )
EGAD_5744_filtered <- EGAD_5744[!(EGAD_5744$subset == "EPCAM+") & (EGAD_5744$Type == "T"),]
EGAD_5744_meta <- data.frame(
  Sample = EGAD_5744_filtered$Sample.Name,
  Study = "EGAD00001005744",
  Subtype = EGAD_5744_filtered$subset,
  Tumor_JuxtaTumor = "tumor",
    directory = "/home/kevin/Documents/PhD/CAF_data/nfcore_results/EGAD00001005744_nfcore_results/star_salmon",
  row.names = 1
)
barkley_samples <- read.csv("/home/kevin/Documents/PhD/rna_seq_bc/metadata/reformat_samples.csv", 
                            header = T, row.names = "samples", check.names = F)
barkley_samples_meta <- data.frame(
  Sample = row.names(barkley_samples),
  Study = "In-House",
  Subtype = "Unknown",
  Tumor_JuxtaTumor = ifelse(barkley_samples$Condition == "Tumour", "tumor", "juxta-tumor"),
  directory = "/home/kevin/Documents/PhD/CAF_data/nfcore_results/inhouse_caf_nfcore_rnaseq_results/star_salmon/",
  row.names = 1
)
metadata <- rbind.data.frame(EGAD_4810_meta, EGAD_3808_meta, 
                             EGAD_6144_meta, EGAD_5744_meta, barkley_samples_meta)
rownames(metadata) <- gsub(pattern = "-", replacement = "\\.", x = rownames(metadata))
# drop directory column
metadata$directory <- NULL
metadata[1:5,]
```

### Prepare transcript annotations

```{r}
#mart <- useMart(biomart = "ensembl", dataset = "hsapiens_gene_ensembl", host="uswest.ensembl.org")
#tx2gene <- getBM(attributes = c("ensembl_transcript_id_version", "hgnc_symbol"), mart = mart, useCache = FALSE)
```

### First attempt at reading in files

```{r get TPM counts}
EGAD00001004810_rds <- readRDS("/home/kevin/Documents/PhD/CAF_data/nfcore_results/EGAD00001004810_nfcore_results/star_salmon/salmon.merged.gene_counts.rds")
EGAD00001004810_tpm <- assays(EGAD00001004810_rds)$abundance
EGAD00001003808_rds <- readRDS("/home/kevin/Documents/PhD/CAF_data/nfcore_results/EGAD00001003808_nfcore_results/star_salmon/salmon.merged.gene_counts.rds")
EGAD00001003808_tpm <- assays(EGAD00001003808_rds)$abundance
EGAD00001006144_rds <- readRDS("/home/kevin/Documents/PhD/CAF_data/nfcore_results/EGAD00001006144_nfcore_results/star_salmon/salmon.merged.gene_counts.rds")
EGAD00001006144_tpm <- assays(EGAD00001006144_rds)$abundance
EGAD00001005744_rds <- readRDS("/home/kevin/Documents/PhD/CAF_data/nfcore_results/EGAD00001005744_nfcore_results/star_salmon/salmon.merged.gene_counts.rds")
EGAD00001005744_tpm <- assays(EGAD00001005744_rds)$abundance
inhouse_rds <- readRDS("/home/kevin/Documents/PhD/CAF_data/nfcore_results/inhouse_caf_nfcore_rnaseq_results/star_salmon/salmon.merged.gene_counts.rds")
inhouse_tpm <- assays(inhouse_rds)$abundance
samps <- c("4810", "3808", "6144", "5744", "inhouse")
dfs <- list(EGAD00001004810_tpm, EGAD00001003808_tpm, EGAD00001006144_tpm, EGAD00001005744_tpm, inhouse_tpm)
```

The inhouse samples were processed with a different version of nf-core/rnaseq, and so have a different number of genes. Here we test the pairwise overlap of the gene names for all samples 

```{r}
mat <- matrix(data = rep(0, length(dfs)*length(dfs)), 
              nrow = length(dfs), ncol = length(dfs), dimnames = list(samps,samps))
mat
for (i in 1:length(dfs)){
  for (j in 1:length(dfs)){
    mat[i,j] <- length(intersect(rownames(dfs[[i]]), rownames(dfs[[j]])))
  }
}
mat
```

We want to have the same genes in each matrix

```{r}
data_combined <- Reduce(function(x, y) merge(x, y, all=TRUE, by="rn", suffixes=c("", ".2")), 
    lapply(dfs, function(x) data.frame(x, rn = row.names(x))))
# get rid of genes with any NA
data_combined <- na.omit(data_combined)
# get rid of genes not expressed in any samples
thresh <- 0.0001
data_combined <- data_combined[rowSums(data_combined[,-1]) > thresh, ]
data_combined <- data_combined %>% remove_rownames %>% column_to_rownames(var="rn")
# get rid of X in colnames
colnames(data_combined) <- gsub(pattern = "X", replacement = "", x = colnames(data_combined))
# only want to keep samples that we want to analyse - those in the metadata object
data_combined <- data_combined[,intersect(rownames(metadata), colnames(data_combined))]
```

```{r}
#data_combined
#summary(data_combined)
avgs <- apply(data_combined, MARGIN = 1, mean)
quants <- quantile(avgs, probs = seq(0,1,0.1))
#length(avgs[avgs > quants[1] & avgs < quants[10]])
# get genes in middle 80 percentile in terms of mean expression in an attempt to remove outliers
genes_80 <- avgs[avgs > quants[1] & avgs < quants[10]]
#genes_80
genes_80_names <- intersect(names(genes_80), rownames(data_combined))
data_combined_filter <- data_combined[genes_80_names,]
#data_combined_filter
```

```{r}

```


For PCA, we want colnames of our `data_combined` object to be the sample names and the rownames of our `metadata` to be the sample names 


```{r}
all(colnames(data_combined) == rownames(metadata))
```

### PCA

```{r}
p <- pca(data_combined, metadata = metadata, removeVar = 0.1)
```
```{r}
biplot(p, colby = 'Study', legendPosition = 'top')
```

```{r}
eigencorplot(p,metavars = c('Study', 'Subtype', 'Tumor_JuxtaTumor'))
```

It is not surprising that the first principal component is correlated with the study of origin. There is also a correlation with subtype, but this could also be due to the link with study - different studies tend to have different subtypes.

Carry out PCA on the filtered data_combined

```{r}
p_filt <- pca(data_combined_filter, metadata = metadata, removeVar = 0.1)
```
```{r}
#biplot(p_filt, colby = 'Study', legendPosition = 'top')
```

```{r}
#hist(data_combined_filter$B86T10)
```

Look for samples with lots of zero expressed genes

```{r}
prop_not_expressed <- c()
ngenes <- nrow(data_combined_filter)
for (i in 1:ncol(data_combined_filter)){
  prop_not_expressed[i] <- length(which(data_combined_filter[,i] == 0))/ngenes
  names(prop_not_expressed)[i] <- colnames(data_combined_filter)[i]
}
range(prop_not_expressed)
quantile(prop_not_expressed)
# remove samples with >90% genes not expressed

```

```{r}
metadata_with_prop_not_expressed <- metadata %>% mutate(prop_not_exp = prop_not_expressed[intersect(names(prop_not_expressed), rownames(metadata))]) %>% 
                                            mutate(ranked_prop_not_exp = rank(prop_not_exp))
#metadata_with_prop_not_expressed
```


```{r}
samples_to_remove <- names(which(prop_not_expressed > 0.9))
data_combined_filter_without_unexpressed_greater_90 <- data_combined_filter %>% dplyr::select(-samples_to_remove)
metadata_remove_outlier <- metadata[!(rownames(metadata) %in% samples_to_remove),] 
p_filt_remove_outlier <- pca(data_combined_filter_without_unexpressed_greater_90, metadata = metadata_remove_outlier, removeVar = 0.1)

```

```{r}
#biplot(p_filt_remove_outlier, colby = 'Subtype', legendPosition = 'top')
```

### Use DeSeq2 to normalise data instead

Difficult to mess around with filters etc.

Plan: rerun in house samples with same nfcore version `3.8.1`
Read in samples with tximport, deseqdataobject etc

```{r}
#data_combined_matrix <- as.matrix(data_combined)
#dds <- Deseqdata
#metadata
```

```{r}
#basedirs <- c("/home/kevin/Documents/PhD/CAF_data/nfcore_results/EGAD00001003808_nfcore_results/star_salmon","/home/kevin/Documents/PhD/CAF_data/nfcore_results/EGAD00001004810_nfcore_results/star_salmon", "/home/kevin/Documents/PhD/CAF_data/nfcore_results/EGAD00001006144_nfcore_results/star_salmon")
#metadata_without_inhouse <- metadata
#files <- file.path(metadata$directory, rownames(metadata), "quant.sf")
```


Next step is to look at batch correction.

## References
