---
title: "CAF subtype analysis"
author: "Kevin Ryan"
date: "`r Sys.time()`"
bibliography: citations.bib
output: 
  github_document:
     toc: true
     toc_depth: 2 
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Introduction

Cancer-associated fibroblasts (CAFs) are a heterogeneous cell type found in the tumour microenvironment. They have a wide array of functions, and tend to be immunosuppressive and cancer-promoting. There have been many attempts to characterise subtypes of CAFs, with much transcriptomic analysis being carried out in the Mechta-Grigoriou lab in Institut Curie. They have identified 4 'subtypes' which can be separated based on the expression of different markers:

 -  S1: FAP^High^, CD29^Med-High^, α^SMAHigh^, PDPN^High^, PDGFRβ^High^
 -  S2: FAP^Neg^, CD29^Low^, αSMANeg-^Low^, PDPN^Low^, PDGFRβ^Low^
 -  S3: FAP^Neg-Low^, CD29^Med^, αSMA^Neg-Low^, PDPN^Low^, PDGFRβ^Low-Med^
 -  S4: FAP^Low-Med^, CD29^High^, αSMA^High^, PDPN^Low^, PDGFRβ^Med^

[@Pelon2020]

FACS gating strategies can be used to isolate these various subtypes. The Mechta-Grigoriou group have done this and have generated bulk RNA-sequencing data for the S1, S3 and S4 subtypes. They generated scRNA-sequencing data for the S1 subtype. This data was deposited on the European Genome Phenome Archive, and was accessed via a Data Transfer Agreement.

The following summarises the data obtained:

+---------------+---------------+-----------------------+-------------------------------+
| Subtype       | Total samples | Studies (Samples)     |  Notes                        |
+===============+===============+=======================+===============================+
| S1            |  28           | - EGAD00001003808 (16)| -  3808 has 12xJuxta-tumor    |
|               |               | - EGAD00001005744 (5) | -  5744 5 samples from LN     |
|               |               | - EGAD00001006144 (7) | -  Sorting vs spreading       |
+---------------+---------------+-----------------------+-------------------------------+
| S2            | 0             | N/A                   | N/A                           |
|               |               |                       |                               |
+---------------+---------------+-----------------------+-------------------------------+
| S3            | 14            | - EGAD00001004810 (14)| -  4810 has 11xJuxta-tumor    |
|               |               |                       | -  Ovarian                    |
+---------------+---------------+-----------------------+-------------------------------+
| S4            | 15            | - EGAD00001003808 (10)| -  3808 has 9xJuxta-tumor     |
|               |               | - EGAD00001005744 (5) | -  5744 5 samples from LN     |
+---------------+---------------+-----------------------+-----------------------+

With the juxta-tumour data, they got tumour and juxta-tumour data from the same patient. However, I have not been able to figure out whether they came from the same patient. Could probably use Optitype to determine HLA allele - match tumour and juxta tumour.

We also have scRNA-seq data for S1.

The data was processed using nf-core/rnaseq version using the default parameters. 

We would expect our tumour-associated normal to be most like the S3 subtype (usually accumulate in juxta-tumours).

## References
