---
title: "GTEx Fibroblast Sources of Variation"
author: "Kevin Ryan"
date: "2023-07-17"
output: html_document
params:
  gtex: ""
  gtex_subject_pheno: ""
  gtex_sample_attributes: ""
  method: ""
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
library(tximeta)
library(DESeq2)
library(vroom)
library(sva)
library(dplyr)
library(stringr)
library(PCAtools)
```

### Convert command-line arguments to variables

```{r}
if (params$method == "nextflow") {
  gtex_file <- params$gtex
  gtex_subject_pheno <- params$gtex_subject_pheno
  gtex_sample_attributes <- params$gtex_sample_attributes
} else {
  gtex_file <- "~/Documents/PhD/subtypes/caf-subtype-analysis/gtex_comparison/data/gene_reads_2017-06-05_v8_cells_cultured_fibroblasts.gct/gene_reads_cells_cultured_fibroblasts.gct"
  gtex_subject_pheno <- "~/Documents/PhD/subtypes/caf-subtype-analysis/gtex_comparison/data/GTEx_Analysis_v8_Annotations_SubjectPhenotypesDS.txt"
  gtex_sample_attributes <- "~/Documents/PhD/subtypes/caf-subtype-analysis/gtex_comparison/data/GTEx_Analysis_v8_Annotations_SampleAttributesDS.txt"
}
sub <- "rstudio"
```

```{r define function}
filter_out_low_expressed <- function(dds, deseq_obj = TRUE){
  library(DESeq2)
  print(paste("no of genes before filtering...", nrow(dds)))
  if (deseq_obj == TRUE){
  # returns a vector of whether the total count of each gene is >= 10 (True or false)
  keep <- rowSums(counts(dds)) >= 10
  # only keep rows (genes) for which keep is TRUE
  dds <- dds[keep,]
  # at least X samples with a count of 10 or more, where X is 5% of samples
  X <- round(0.05*ncol(dds))
  keep <- rowSums(counts(dds) >= 10) >= X
  dds <- dds[keep,]
  } else {
    # returns a vector of whether the total count of each gene is >= 10 (True or false)
  keep <- rowSums(dds) >= 10
  # only keep rows (genes) for which keep is TRUE
  dds <- dds[keep,]
  # at least X samples with a count of 10 or more, where X is 5% of samples
  X <- round(0.05*ncol(dds))
  keep <- rowSums(dds >= 10) >= X
  dds <- dds[keep,]
  }
  print(paste("no of genes after filtering...", nrow(dds)))
  return(dds)
}
```

## Question: what are the main sources of variation in the GTEx Fibroblast samples?

### Read in and preprocess data

```{r}
gtex_in <- vroom(gtex_file, skip = 2)
gtex_in <- gtex_in %>% dplyr::select(-c(id, Description))
gtex_in <- as.data.frame(gtex_in)
rownames(gtex_in) <- gtex_in$Name
gtex_in$Name <- NULL
gtex_in <- filter_out_low_expressed(gtex_in, deseq_obj = F)
gtex_in$Gene <- str_split_fixed(rownames(gtex_in), pattern = "\\.", n = 2)[,1]
gtex_in[1:5,1:5]
```

```{r}
get_median_duplicated_genenames <- function(df){
  n_occur <- data.frame(table(df$Gene))
  dups <- df[df$Gene %in% n_occur$Var1[n_occur$Freq > 1],]
  not_dups <- df[!(df$Gene %in% n_occur$Var1[n_occur$Freq > 1]),]
  dups_summarise <- dups %>% group_by(Gene) %>% dplyr::summarise((across(where(is.numeric), median)))
  df_not_dup <- rbind.data.frame(not_dups, dups_summarise)
  return(df_not_dup)
}
gtex_in_not_dup <- get_median_duplicated_genenames(gtex_in)
```

#### read in gtex subject-level metadata

```{r}
gtex_subject_pheno_in <- vroom(gtex_subject_pheno)
gtex_subject_pheno_in <- data.frame(gtex_subject_pheno_in)
gtex_in[1:5,1:5]
gtex_fibroblast_subjid <- paste("GTEX", str_split_fixed(colnames(gtex_in)[-ncol(gtex_in)], pattern = "-", n = 5)[,2], sep = "-")
rownames(gtex_subject_pheno_in) <- gtex_subject_pheno_in$SUBJID
gtex_subject_pheno_in_ord <- gtex_subject_pheno_in[gtex_fibroblast_subjid,]
colnames(gtex_subject_pheno_in_ord)[1] <- "names"
```

#### read in gtex sample-level metadata

```{r}
gtex_sample_pheno_in <- data.frame(vroom(gtex_sample_attributes))
rownames(gtex_sample_pheno_in) <- gtex_sample_pheno_in$SAMPID
gtex_sample_pheno_in_fibroblast <- gtex_sample_pheno_in[colnames(gtex_in)[-ncol(gtex_in)],]
gtex_fibroblast_subjid2 <- paste("GTEX", str_split_fixed(rownames(gtex_sample_pheno_in_fibroblast), pattern = "-", n = 5)[,2], sep = "-")
rownames(gtex_sample_pheno_in_fibroblast) <- gtex_fibroblast_subjid2
# put in same order as subject-level metadata
gtex_sample_pheno_in_fibroblast_ord <- gtex_sample_pheno_in_fibroblast[gtex_fibroblast_subjid,]
gtex_sample_pheno_in_fibroblast_ord$SAMPID <- rownames(gtex_sample_pheno_in_fibroblast_ord)
colnames(gtex_sample_pheno_in_fibroblast_ord)[1] <- "names"
```

#### combine the two

```{r}
gtex_sample_pheno_in_complete <- full_join(gtex_subject_pheno_in_ord, gtex_sample_pheno_in_fibroblast_ord, by = "names")
```

```{r}
gtex_sample_pheno_in_complete
```

#### get gene expression subjects in right order

```{r}
gtex_fibroblast_subjid3 <- paste("GTEX", str_split_fixed(colnames(gtex_in_not_dup), pattern = "-", n = 5)[,2], sep = "-")
colnames(gtex_in_not_dup) <- gtex_fibroblast_subjid3
gtex_in_not_dup_ord <- gtex_in_not_dup[,gtex_fibroblast_subjid]
```

#### create dds object

```{r}
not_all_na <- function(x) any(!is.na(x))
not_any_na <- function(x) all(!is.na(x))
gtex_sample_pheno_in_complete <- gtex_sample_pheno_in_complete %>% dplyr::select(where(not_any_na))
gtex_sample_pheno_in_complete_eigencor <- gtex_sample_pheno_in_complete %>% dplyr::select(names, where(is.numeric), -c(SMUNMPRT, SMRDLGTH, SMUNPDRD, SMESTLBS, SMDPMPRT), SMCENTER, SMNABTCH, SMNABTCHD, SMGEBTCH, SMGEBTCHD)
gtex_sample_pheno_in_complete_char <- gtex_sample_pheno_in_complete %>% dplyr::select(names, where(is.character), where(is.factor))
                                                                                      #AGE, SEX, SMCENTER, SMTSD,SMUBRID )
#gtex_sample_pheno_in_complete <- mutate_if(gtex_sample_pheno_in_complete, is.character, as.factor)
dds <- DESeqDataSetFromMatrix(countData = gtex_in_not_dup_ord, colData = gtex_sample_pheno_in_complete, design = ~1)
vsd <- vst(dds, blind = TRUE)
```

```{r}
gtex_sample_pheno_in_complete
```

```{r}
p <- pca(assay(vsd), metadata = colData(dds))
```

```{r}
peigencor <- suppressWarnings(eigencorplot(p,
    components = getComponents(p, 1:10),
    metavars = colnames(gtex_sample_pheno_in_complete_eigencor),
    col = c('white', 'cornsilk1', 'gold', 'forestgreen', 'darkgreen'),
    cexCorval = 0.7,
    colCorval = 'black',
    fontCorval = 2,
    posLab = 'bottomleft',
    rotLabX = 45,
    posColKey = 'top',
    cexLabColKey = 1.5,
    scale = TRUE,
    corFUN = 'pearson',
    corUSE = 'pairwise.complete.obs',
    corMultipleTestCorrection = 'none',
    main = 'PCs Correlation with Numeric Covariates',
    cexMain = 1.3,
    colFrame = 'white',
    plotRsquared = TRUE))
```

```{r}
peigencor
```

Covariates correlated with PC1:
- SMSPLTRD - Split Reads: The number of reads that span an exon-exon boundary
- SMRRNART - rRNA Rate: Ratio of all reads aligned to rRNA regions to total reads
- SMNABTCH - Nucleic Acid Isolation Batch ID

Numeric covariates correlated with PC2:
- SMRRNART - rRNA Rate: Ratio of all reads aligned to rRNA regions to total reads

Now look at categorical covariates:

```{r}
gtex_sample_pheno_in_complete_char
```

```{r}
data <- plotPCA(vsd, intgroup = c("AGE", "SEX", "SMCENTER", "SMNABTCH", "SMNABTCHD", "SMGEBTCH", "SMGEBTCHD", "SMSPLTRD"), 
                ntop = nrow(vsd), returnData = TRUE)
data$SEX <- gsub(1, "Male", data$SEX)
data$SEX <- gsub(2, "Female", data$SEX)

percentVar <- round(100 * attr(data, "percentVar"))
ggplot(data, aes(PC1, PC2,  colour=as.character(SEX))) +   
  geom_point(size=3) +
 xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance"))
```

```{r}
plot_splitreads <- ggplot(data, aes(PC1, PC2,  colour=log(SMSPLTRD))) +   
  geom_point(size=3) +
 xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance")) +
  ggtitle("PCA biplot coloured by number of split reads GTEx")
ggsave(plot_splitreads, filename = "~/Documents/PhD/subtypes/caf-subtype-analysis/gtex_comparison/results/pca_splitreads.png")

```

```{r}
pca_sex <- ggplot(data, aes(PC1, PC2,  colour=SEX)) +   
  geom_point(size=3) +
 xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance")) +
  ggtitle("PCA biplot coloured by sex GTEX only")
ggsave(pca_sex, filename = "~/Documents/PhD/subtypes/caf-subtype-analysis/gtex_comparison/results/pca_before_correction_sex.png")
```

```{r}
pca_age <- ggplot(data, aes(PC1, PC2,  colour=AGE)) +   
  geom_point(size=3) +
 xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance")) +
  ggtitle("PCA biplot coloured by age GTEX only")
ggsave(pca_age, filename = "~/Documents/PhD/subtypes/caf-subtype-analysis/gtex_comparison/results/pca_before_correction_age.png")
```

```{r}
# plot_pca <- function(pca_obj, by){
#   pca_obj$fun.y <- pca_obj[, by]
#   if (is.numeric(colData(pca_obj)$`by`)){
#     colData(pca_obj)$`by` <- as.numeric(colData(pca_obj)$`by`)
#   }
#   percentVar <- round(100 * attr(pca_obj, "percentVar"))
#   ggplot(pca_obj, aes(PC1, PC2,  colour=by)) +   
#   geom_point(size=3) +
#  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
#   ylab(paste0("PC2: ",percentVar[2],"% variance"))
# }
# 
# plot_pca(data , by = "SMUNMPRT")
```

```{r}
ggplot(data, aes(PC1, PC2,  colour=SMNABTCH)) +   
  geom_point(size=3) +
 xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance"))
```

```{r}
table(gtex_sample_pheno_in_complete_char$SMNABTCH)
```

```{r}
ggplot(data, aes(PC1, PC2,  colour=SMCENTER)) +   
  geom_point(size=3) +
 xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance"))
```

```{r}
ggplot(data, aes(PC1, PC2,  colour=SMNABTCH)) +   
  geom_point(size=3) +
 xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance"))
length(table(colData(vsd)$SMNABTCH))
table(colData(vsd)$SMNABTCH)
length(colData(vsd)$SMNABTCH)
```

```{r}

ggplot(data, aes(PC1, PC2,  colour=SMNABTCHD)) +   
  geom_point(size=3) +
 xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance"))
```

```{r}
ggplot(data, aes(PC1, PC2,  colour=SMGEBTCH)) +   
  geom_point(size=3) +
 xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance"))
```

```{r}
ggplot(data, aes(PC1, PC2,  colour=SMGEBTCHD)) +   
  geom_point(size=3) +
 xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance"))
```

try combat-seq removing SMNABTCH batch effect

```{r}
# dds.mat <- assay(dds)
# batch <- colData(dds)$SMNABTCH
# print(paste("Number of genes before filtering: ", nrow(dds.mat), sep = ""))
# # remove genes with 0 in > 1/3 of samples as per GitHub user benostendorf https://github.com/zhangyuqing/ComBat-seq/issues/20 
# dds.mat.filtered <- dds.mat[apply(dds.mat, 1, function(x) sum(x == 0)) < ncol(dds.mat) / 3, ]
# print(paste("Number of genes after filtering: ", nrow(dds.mat.filtered), sep = ""))
# print("batch correcting...")
# ptm <- proc.time()
# dds.mat.batch.corrected <- ComBat_seq(counts = dds.mat.filtered, batch = batch, full_mod = FALSE, group=NULL)
# time_taken <- proc.time() - ptm
# print("time taken...")
# print(time_taken)
# #dds.remove.outliers.mat.batch.corrected
# # for some reason started to get the na values due to too large numbers error here, correcting by dividing everything by 4 20230517
# dds.batch.corrected <- DESeqDataSetFromMatrix(round(dds.mat.batch.corrected/4), colData = colData(dds_combined), design = ~1)
# vsd.batch.corrected <- vst(dds.batch.corrected, blind = TRUE)
```

There are numerous sources of unwanted variation, too many to take into account all together. SVA can be used to remove unwanted sources of variation, without having to specify the actual batches. Combat-seq would be useful to try 

```{r}
batch <- colData(dds)$SMNABTCH
design0 <- model.matrix(~gtex_sample_pheno_in_complete$AGE)
vsd_batch_correct <- vsd
assay(vsd_batch_correct) <- removeBatchEffect(assay(vsd), batch = batch, design=design0)
```

```{r}
data2 <- plotPCA(vsd_batch_correct, intgroup = c("AGE", "SEX", "SMCENTER", "SMNABTCH", "SMNABTCHD", "SMGEBTCH", "SMGEBTCHD"), 
                ntop = nrow(vsd_batch_correct), returnData = TRUE)
percentVar <- round(100 * attr(data2, "percentVar"))
ggplot(data2, aes(PC1, PC2, color = as.character(SEX))) +   
  geom_point(size=3) +
 xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance"))
```

```{r}
ggplot(data2, aes(PC1, PC2, color = SMNABTCH)) +   
  geom_point(size=3) +
 xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance"))
```

```{r}
p <- pca(assay(vsd_batch_correct), metadata = colData(dds))
```

```{r}
peigencor <- suppressWarnings(eigencorplot(p,
    components = getComponents(p, 1:10),
    metavars = colnames(gtex_sample_pheno_in_complete_eigencor),
    col = c('white', 'cornsilk1', 'gold', 'forestgreen', 'darkgreen'),
    cexCorval = 0.7,
    colCorval = 'black',
    fontCorval = 2,
    posLab = 'bottomleft',
    rotLabX = 45,
    posColKey = 'top',
    cexLabColKey = 1.5,
    scale = TRUE,
    corFUN = 'pearson',
    corUSE = 'pairwise.complete.obs',
    corMultipleTestCorrection = 'none',
    main = 'PCs Correlation with Numeric Covariates',
    cexMain = 1.3,
    colFrame = 'white',
    plotRsquared = TRUE))
```

```{r}
peigencor
```

